From 00e44e8ed7bc36334066253159a084b0d3b492d0 Mon Sep 17 00:00:00 2001
From: Jakub Adam <jakub.adam@collabora.com>
Date: Tue, 25 Aug 2020 19:12:13 +0200
Subject: [PATCH] srtobject: post a message on the bus when broken socket is
 detected

So that the application gets notified may react to it.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-plugins-bad/-/merge_requests/1935>
---
 ext/srt/gstsrtobject.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/ext/srt/gstsrtobject.c b/ext/srt/gstsrtobject.c
index 842a6e59a..3439f319a 100644
--- a/ext/srt/gstsrtobject.c
+++ b/ext/srt/gstsrtobject.c
@@ -1638,8 +1638,9 @@ gst_srt_object_write_one (GstSRTObject * srtobject,
       case SRTS_BROKEN:
       case SRTS_NONEXIST:
       case SRTS_CLOSED:
-        GST_WARNING_OBJECT (srtobject->element,
-            "Invalid SRT socket. Trying to reconnect");
+        GST_ELEMENT_WARNING (srtobject->element, RESOURCE, WRITE, NULL,
+            ("Invalid SRT socket. Trying to reconnect. (%s)",
+                srt_getlasterror_str ()));
         gst_srt_object_close (srtobject);
         if (!gst_srt_object_open_internal (srtobject, cancellable, error)) {
           return -1;
-- 
2.25.1

